cpp/src/parquet/chunker_internal_test.cc

```diff
@@ -0,0 +1,1397 @@
+  ASSERT_EQ(inserted->num_rows(), base->num_rows() + edit2_->num_rows());
+
+  ASSERT_OK_AND_ASSIGN(
+      auto base_info,
+      WriteAndGetParquetInfo(base, kMinChunkSize, kMaxChunkSize, kEnableDictionary,
+                             ParquetDataPageVersion::V1, kRowGroupLength));
+  ASSERT_OK_AND_ASSIGN(
+      auto inserted_info,
+      WriteAndGetParquetInfo(inserted, kMinChunkSize, kMaxChunkSize, kEnableDictionary,
+                             ParquetDataPageVersion::V1, kRowGroupLength));
+
+  ASSERT_EQ(base_info.size(), 7);
+  ASSERT_EQ(inserted_info.size(), 7);
+
+  ASSERT_EQ(base_info.at(0).page_lengths, inserted_info.at(0).page_lengths);
+  ASSERT_EQ(base_info.at(1).page_lengths, inserted_info.at(1).page_lengths);
+  for (size_t i = 2; i < inserted_info.size() - 1; i++) {
+    AssertPageLengthDifferences(base_info.at(i), inserted_info.at(i),
+                                /*exact_number_of_equal_diffs=*/0,
+                                /*exact_number_of_larger_diffs=*/1,
+                                /*exact_number_of_smaller_diffs=*/1, edit2_->num_rows());
+  }
+  AssertPageLengthDifferences(base_info.back(), inserted_info.back(),
+                              /*exact_number_of_equal_diffs=*/0,
+                              /*exact_number_of_larger_diffs=*/1,
+                              /*exact_number_of_smaller_diffs=*/0, edit2_->num_rows());
+}
+
+TEST_F(TestCDCMultipleRowGroups, DeleteOnce) {
+  auto constexpr kRowGroupLength = 128 * 1024;
+  auto constexpr kEnableDictionary = false;
+  auto constexpr kMinChunkSize = 0 * 1024;
+  auto constexpr kMaxChunkSize = 128 * 1024;
+
+  ASSERT_OK_AND_ASSIGN(auto base,
+                       ConcatAndCombine({part1_, edit1_, part2_, part3_, edit2_}));
+  ASSERT_OK_AND_ASSIGN(auto deleted, ConcatAndCombine({part1_, part2_, part3_, edit2_}));
+  ASSERT_FALSE(base->Equals(*deleted));
+  ASSERT_EQ(deleted->num_rows(), base->num_rows() - edit1_->num_rows());
+
+  ASSERT_OK_AND_ASSIGN(
+      auto base_info,
+      WriteAndGetParquetInfo(base, kMinChunkSize, kMaxChunkSize, kEnableDictionary,
+                             ParquetDataPageVersion::V1, kRowGroupLength));
+  ASSERT_OK_AND_ASSIGN(
+      auto deleted_info,
+      WriteAndGetParquetInfo(deleted, kMinChunkSize, kMaxChunkSize, kEnableDictionary,
+                             ParquetDataPageVersion::V1, kRowGroupLength));
+
+  ASSERT_EQ(base_info.size(), 7);
+  ASSERT_EQ(deleted_info.size(), 7);
+
+  ASSERT_EQ(base_info.at(0).page_lengths, deleted_info.at(0).page_lengths);
+  ASSERT_EQ(base_info.at(1).page_lengths, deleted_info.at(1).page_lengths);
+  for (size_t i = 2; i < deleted_info.size() - 1; i++) {
+    AssertPageLengthDifferences(base_info.at(i), deleted_info.at(i),
+                                /*exact_number_of_equal_diffs=*/0,
+                                /*exact_number_of_larger_diffs=*/1,
+                                /*exact_number_of_smaller_diffs=*/1, edit1_->num_rows());
```

Comment:
Why a larger diff _and_ a smaller diff? It seems like you're just using magic values to satisfy the actual results.

Reply:
Haha, no :) 
I will try to give some description to those cases.

Reply:
Added comments to all these scenarios.